apm-agent-plugins/apm-jaxrs-plugin/src/main/java/co/elastic/apm/agent/jaxrs/JaxRsTransactionNameInstrumentation.java

```diff
@@ -54,18 +57,25 @@
 
     private final Collection<String> applicationPackages;
     private final JaxRsConfiguration configuration;
+    @Nullable
+    @VisibleForAdvice
+    public static JaxRsTransactionHelper jaxRsTransactionHelper;
```

Comment:
You can give up the helper and store a static reference to the WebConfiguration instead

Reply:
deleted

Reply:
with
```
private static WebConfiguration webConfiguration;

    public JaxRsTransactionNameInstrumentation(ElasticApmTracer tracer) {
        applicationPackages = tracer.getConfig(StacktraceConfiguration.class).getApplicationPackages();
        configuration = tracer.getConfig(JaxRsConfiguration.class);
        elasticApmTracer = tracer;
        webConfiguration = tracer.getConfig(WebConfiguration.class);
    }

    @Advice.OnMethodEnter(suppress = Throwable.class)
    private static void setTransactionName(@SimpleMethodSignature String signature,
                                           @JaxRsPath @Nullable String pathAnnotationValue) {
        if (tracer != null) {
            final Transaction transaction = tracer.currentTransaction();
            if (transaction != null) {
                String transactionName = signature;
                if (webConfiguration.isUseAnnotationValueForTransactionName()) {
                    if (pathAnnotationValue != null) {
                        transactionName = pathAnnotationValue;
                    }
                }
                transaction.withName(transactionName);
            }
        }
    }

```
I got access exception :
```
[main] DEBUG co.elastic.apm.agent.impl.ElasticApmTracer - Activating '' 00-f21a1ff4279a52fa30b11c0f444bea5f-cf2ac1310b4b3e96-01 (18137eab) on thread 1
java.lang.IllegalAccessError: tried to access field co.elastic.apm.agent.jaxrs.JaxRsTransactionNameInstrumentation.webConfiguration from class co.elastic.apm.agent.jaxrs.JaxRsTransactionNameInstrumentationTest$ResourceWithPath
	at co.elastic.apm.agent.jaxrs.JaxRsTransactionNameInstrumentationTest$ResourceWithPath.testMethod(JaxRsTransactionNameInstrumentationTest.java:326)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)

```

Reply:
The `@OnMethodEnter` code is injected into a class, it does not run from where it is written as an inner class, so you can't make this field `private`- it needs to be `public`.

Currently, you added the `use_path_template_as_transaction_name` as non-dynamic, so you can store a `public static boolean` for that instead of the WebConfiguration:
```
    private static boolean useAnnotationValueForTransactionName;
    ...
    public JaxRsTransactionNameInstrumentation(ElasticApmTracer tracer) {
        applicationPackages = tracer.getConfig(StacktraceConfiguration.class).getApplicationPackages();
        configuration = tracer.getConfig(JaxRsConfiguration.class);
        elasticApmTracer = tracer;
        useAnnotationValueForTransactionName = tracer.getConfig(WebConfiguration.class).isUseAnnotationValueForTransactionName();
    }
```

Reply:
replaced with 
```
    public static boolean useAnnotationValueForTransactionName;
```